### DISCUSSION TITLE
Next.JS server component stuck on loading

### DISCUSSION THREAD

**Discussion by Jwiggiff** [vercel/community#2593] [Help]
### Summary

I've written a Next.JS app where one of the pages makes an API request. When I deploy to Vercel, it indicates that this page is a function. However, when I go to the page, it stays on the `loading.tsx` component. In the Vercel logs, it shows the request with all the info on the Serverless Function that ran, yet nothing comes up on the page. After a while I get the below error in the console. It works completely fine locally.

### Additional information

```
Uncaught (in promise) Error: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received
```

### Example

Link to the repo:
https://github.com/Jwiggiff/infinipedia

**Comment by amyegan** [comment]
Hi @Jwiggiff. If you can share your public repo or a minimal reproduction, someone here might be able to recognize what is going wrong.

**Reply by Jwiggiff** [reply]
Hi @amyegan I've edited my original post to include this. Thanks!
